app/src/main/java/com/powerhungers/fomezero/feature/producer/basket/presentation/BasketViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package com.powerhungers.fomezero.feature.producer.basket.presentation
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.powerhungers.fomezero.common.exception.EmptyCheckboxException
+import com.powerhungers.fomezero.common.extension.postError
+import com.powerhungers.fomezero.common.extension.postLoading
+import com.powerhungers.fomezero.common.extension.postNeutral
+import com.powerhungers.fomezero.common.extension.postSuccess
+import com.powerhungers.fomezero.common.utils.ViewState
+import com.powerhungers.fomezero.feature.producer.basket.domain.usecase.BasketUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.launch
+
+class BasketViewModel(private val basketUseCase: BasketUseCase) : ViewModel() {
+
+    private val basketViewState = MutableLiveData<ViewState<Unit>>()
+    val basketLiveData = basketViewState as LiveData<ViewState<Unit>>
+
+    fun basket(smallBasket: Boolean, mediumBasket: Boolean, bigBasket: Boolean) {
+        viewModelScope.launch {
+            basketUseCase(smallBasket, mediumBasket, bigBasket)
+                    .flowOn(Dispatchers.IO)
+                    .onStart { basketViewState.postLoading() }
+                    .catch { basketViewState.postError(EmptyCheckboxException()) }
```

Comment:
dessa forma vc esta afirmando que todo vez que tivermos um erro sempre sera retornado pra view empty checkbox exception, se o datasource jogar uma excption diferente, nunca iremos saber, alem disso anular a regra que vc fez no use case
```suggestion
                    .catch { basketViewState.postError(it)) }
```